cmd/tink-cli/cmd/hardware/ip.go

```diff
@@ -28,11 +29,15 @@ var ipCmd = &cobra.Command{
 	},
 	Run: func(cmd *cobra.Command, args []string) {
 		for _, ip := range args {
-			hw, err := client.HardwareClient.ByIP(context.Background(), &hardware.GetRequest{IP: ip})
+			hw, err := client.HardwareClient.ByIP(context.Background(), &hardware.GetRequest{Ip: ip})
```

Comment:
Is there a way for the generated code to influence the name for the generated struct fields? Looking at `Ip` -> `IP` and `Mac` -> `MAC` and maybe others?

Reply:
I didn't really look into it too much since it seemed to work fine with boots without any issue. Is there a reason/necessity for the names to be capitalized? I could change them back to uppercase in the proto if needed

Reply:
its a code nit, but acronyms should be in all caps. It also matches the types in the stdlib so not being an oddball would be good.

Reply:
okay, I'll change it back then

Reply:
I remember why I decided on all lowercase, it's because there are problems when the message name is the same as the field name. For example:
```
message RAID {
	string name = 1;
	string level = 2;
	repeated string devices = 3;
	int64 spare = 4;
}

repeated RAID RAID = 5;
```
this is giving me the error:
```
Generating hardware.pb.go...
hardware/hardware.proto:164:41: "RAID" is already defined in "github.com.tinkerbell.tink.protos.hardware.Hardware.Metadata.Instance.Storage".
hardware/hardware.proto:175:42: "RAID" is not defined.
```
So I decided on all lowercase for the field name (which is also the convention for protobuf).  
I could still rename the message name to something like "Raid", so I guess it all boils down to which convention to follow (go's or protobuf's). Which one should I go with?

Reply:
ah ok, sigh. lets go with protobuf then. I'm curious is there an equivalent of `json:"SomeKeyName"` metadata we can add so we can both?

Reply:
looks like no. https://github.com/golang/protobuf/issues/555. In that case lets stay with protobuf norms. Transforming protoc generated structs into human readable/meaningful structs is something that I think always makes sense and we could address then.

Reply:
Hmm what do you mean by have both? like in terms of how the json is marshaled/unmarshalled?

Reply:
okay, gotcha